### ISSUE TITLE
Can exchndl be made to catch RaiseFailFastException?

### ISSUE THREAD

**Issue by alvinhochun** [jrfonseca/drmingw#64]
Qt decided that its [fatal error handler should call `RaiseFailFastException`](https://invent.kde.org/qt/qt/qtbase/-/blob/v5.12.12/src/corelib/global/qlogging.cpp#L1885-1899) instead of just `abort` ([context](https://bugreports.qt.io/browse/QTBUG-34460)). [This issue](https://github.com/MicrosoftDocs/cpp-docs/issues/219) also touches on the exceptional behaviour of `abort` which Qt is trying to prevent.

However, `exchndl` is apparently unable to dump the stack trace when the Qt fatal error handler is called. Are there any tricks to make `exchndl` work for this case?

**Comment by jrfonseca** [comment]
`RaiseFailFastException` specifically skips any exception handlers.  So I see the following options (from simplest to hardest):

1. The ideal would be for Qt to provide a callback from `qt_message_fatal`, where one could just raise a normal exception which would be trapped by ExcHndl.
2. Catch exception out of process, through the debugging API.  That is, use DrMinGW's [catchsegv](https://github.com/jrfonseca/drmingw#catchsegv) instead of ExcHndl.  There might be some performance impact, particularly initialization (when loading all DLLs.)  Give it a try running Krita inside (`catchsegv -m krita ...`) -- if performance is acceptable, then you could have setup krita to always be run inside catchsegv (via the shortcuts, or through a launcher.) 
3. Use [Detours](https://github.com/microsoft/Detours) inside ExcHndl to intercept `RaiseFailFastException` and report the exception.  In one way, this is less disruptive, but potentially more brittle, as these code patching mechanisms are the same used by malware, hence are constantly getting constrained.

I hope this helps.

**Comment by alvinhochun** [comment]
Thanks for the suggestions. Given these options I think I will patch Qt's behaviour for now and consider moving to a catchsegv-like approach in the future. I've always wanted to implement a better UI on crash, so maybe I will try to make a fancier GUI version of it (not now though 😆 ). I tried catchsegv on Krita and didn't see any slowdown on initialization... but frankly Krita on Windows already starts up way slower than on Linux (10 seconds vs 3 seconds).

Side note: When I tested with catchsegv, it didn't seem to be able to produce line number info (exchndl outputs them fine). I have split debug file linked with gnu-debuglink, maybe that is broken with catchsegv?
